src/manager.cpp

```diff
@@ -178,6 +294,127 @@ std::string
+                     kwdVal->size());
+    }
+
+    // get the first part of expanded location code to check for FC or TM
+    std::string l_firstKwd = i_expandedLocationCode.substr(1, 4);
+
+    std::string o_unexpandedLocationCode{};
+    uint16_t o_nodeNummber = constants::INVALID_NODE_NUMBER;
+
+    // check if this value matches the value of FC kwd
+    if (l_kwd.substr(0, 4) == l_firstKwd)
+    {
+        // period(.) should be there in expanded location code to seggregate FC,
+        // Node number and SE values.
+        size_t l_nodeStartPos = i_expandedLocationCode.find('.');
+        if (l_nodeStartPos == std::string::npos)
+        {
+            phosphor::logging::elog<types::InvalidArgument>(
+                types::Argument::ARGUMENT_NAME("LOCATIONCODE"),
+                types::Argument::ARGUMENT_VALUE(
+                    i_expandedLocationCode.c_str()));
+        }
+
+        size_t l_nodeEndPos = i_expandedLocationCode.find('.',
+                                                          l_nodeStartPos + 1);
+        if (l_nodeEndPos == std::string::npos)
+        {
+            phosphor::logging::elog<types::InvalidArgument>(
+                types::Argument::ARGUMENT_NAME("LOCATIONCODE"),
+                types::Argument::ARGUMENT_VALUE(
+                    i_expandedLocationCode.c_str()));
+        }
+
+        // skip 3 for '.ND'
+        o_nodeNummber = std::stoi(i_expandedLocationCode.substr(
+            l_nodeStartPos + 3, (l_nodeEndPos - l_nodeStartPos - 3)));
+
+        // confirm if there are other details apart FC, Node number and SE in
+        // location code
+        if (i_expandedLocationCode.length() >
+            constants::EXP_LOCATION_CODE_MIN_LENGTH)
+        {
+            o_unexpandedLocationCode =
+                i_expandedLocationCode[0] + (std::string) "fcs" +
+                i_expandedLocationCode.substr(l_nodeEndPos + 1 +
+                                                  constants::SE_KWD_LENGTH,
+                                              std::string::npos);
+        }
+        else
+        {
+            o_unexpandedLocationCode = "Ufcs";
+        }
+    }
+    else
+    {
+        // read TM kwd value
+        l_retVal = utils::readDbusProperty(
+            "xyz.openbmc_project.Inventory.Manager",
+            "/xyz/openbmc_project/inventory/system/chassis/motherboard",
+            "com.ibm.ipzvpd.VSYS", "TM");
+
+        if (auto kwdVal = std::get_if<types::BinaryVector>(&l_retVal))
+        {
+            l_kwd.assign(reinterpret_cast<const char*>(kwdVal->data()),
+                         kwdVal->size());
+        }
+
+        // check if the substr matches to TM kwd
+        if (l_kwd.substr(0, 4) == l_firstKwd)
+        {
+            // system location code will not have any other details and node
+            // number
+            o_unexpandedLocationCode = "Umts";
+        }
+        // it does not belong to either "fcs" or "mts"
```

Comment:
The give location code is neither fcs nor mts. ?

Reply:
Updated